NPE in creating workspaces

Caused by: java.lang.NullPointerException
	at fi.otavanopisto.muikku.plugins.schooldatapyramus.PyramusWorkspaceSchoolDataBridge.createWorkspaceEntity(PyramusWorkspaceSchoolDataBridge.java:353)

--- a/muikku-school-data-pyramus/src/main/java/fi/otavanopisto/muikku/plugins/schooldatapyramus/PyramusWorkspaceSchoolDataBridge.java
+++ b/muikku-school-data-pyramus/src/main/java/fi/otavanopisto/muikku/plugins/schooldatapyramus/PyramusWorkspaceSchoolDataBridge.java
@@ -350,12 +350,12 @@ public class PyramusWorkspaceSchoolDataBridge implements WorkspaceSchoolDataBrid
             String.format("/courses/courses/%d/educationTypes/%d/educationSubtypes", course.getId(), courseEducationType.getId()),
             CourseEducationSubtype[].class);
         
-        if (educationSubtypeIdentifier == null && courseEducationSubtypes.length == 1) {
-          educationSubtypeIdentifier = identifierMapper.getEducationSubtypeIdentifier(courseEducationSubtypes[0].getEducationSubtypeId());
-        }
-        
         if (courseEducationSubtypes == null) {
           continue;
+        }
+
+        if (educationSubtypeIdentifier == null && courseEducationSubtypes.length == 1) {
+          educationSubtypeIdentifier = identifierMapper.getEducationSubtypeIdentifier(courseEducationSubtypes[0].getEducationSubtypeId());
         }
         
         EducationType educationType = pyramusClient.get(

